File: cpp/program/manager.cpp
/*
 * LICENSE: All Rights Reserved
 * (c) 2025 Willsoon Tulus P. Simanjuntak
 * (c) 2025 Universitas Pendidikan Indonesia
 *
 * Saya Willsoon Tulus Parluhutan Simanjuntak mengerjakan evaluasi
 * Tugas Praktikum 3 dalam mata kuliah Desain Pemrograman
 * Berorientasi Objek untuk keberkahanNya maka saya tidak
 * melakukan kecurangan seperti yang telah dispesifikasikan.
 * Aamiin.
 */

/* Deklarasi header custom */
#include "header.hpp"

/* Deklarasi kelas */

// Deklarasi konstruktor awal untuk inisialisasi
Manager::Manager() {
    this->kode = "";
    this->bidang = "";
    this->periode = "";
};

// Konstruktor untuk membuat objek baru oleh fungsi luar
Manager::Manager(string nik, string nama, string nomorHP, 
                 string kode, string bidang, string periode) 
                 : Penduduk(nik, nama, nomorHP) 
{
    this->kode = kode;
    this->bidang = bidang;
    this->periode = periode;
};

// Fungsi setter untuk menentukan kode manager
void Manager::setKode(string kode) {
    this->kode = kode;
};

// Fungsi setter untuk menentukan bidang manager
void Manager::setBidang(string bidang) {
    this->bidang = bidang;
};

// Fungsi setter untuk menentukan periode manager
void Manager::setPeriode(string periode) {
    this->periode = periode;
};

// Fungsi getter untuk mendapatkan kode manager
string Manager::getKode() {
    return this->kode;
};

// Fungsi getter untuk mendapatkan bidang manager
string Manager::getBidang() {
    return this->bidang;
};

// Fungsi getter untuk mendapatkan periode manager
string Manager::getPeriode() {
    return this->periode;
};

// Deklarasi destruktor
Manager::~Manager() {};

File: python/program/manager.py
# LICENSE: All Rights Reserved
# (c) 2025 Willsoon Tulus P. Simanjuntak
# (c) 2025 Universitas Pendidikan Indonesia
#
# Saya Willsoon Tulus Parluhutan Simanjuntak mengerjakan evaluasi
# Tugas Praktikum 3 dalam mata kuliah Desain Pemrograman
# Berorientasi Objek untuk keberkahanNya maka saya tidak
# melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

# Deklarasi dependency #

from penduduk import Penduduk

# Deklarasi kelas objek #

class Manager(Penduduk):

    # constructor untuk inisialisasi atribut

    def __init__(self, nik, nama, nomorHP, kode, bidang, periode):
        super().__init__(nik, nama, nomorHP)
        self._kode = kode
        self._bidang = bidang
        self._periode = periode

    # Fungsi setter (write atribut)
    def setKode(self, kode: str) -> None:
        self._kode = str(kode)

    def setBidang(self, bidang: str) -> None:
        self._bidang = str(bidang)

    def setPeriode(self, periode: str) -> None:
        self._periode = str(periode)

    # Fungsi getter (read atribut)
    def getKode(self) -> str:
        return self._kode

    def getBidang(self) -> str:
        return self._bidang

    def getPeriode(self) -> str:
        return self._periode
